src/auth/downloader.cpp

```diff
@@ -0,0 +1,111 @@
+#include "downloader.h"
+
+#include <QtNetwork>
+
+Downloader::Downloader(QObject* pobj) : QObject(pobj)
+{   //qDebug() << "Downloader";
+    mNetworkAccessManager = new QNetworkAccessManager(this);
+    connect(mNetworkAccessManager, &QNetworkAccessManager::finished, this,   &Downloader::onFinished);
+    //qDebug() << "/Downloader";
+}
+
+void Downloader::downloadGET(const QUrl &url, const QString &userName,
+                             const QString &tokenOrPassword)
+{   //qDebug() << "Downloader::downloadGET";
+    QNetworkRequest request(url);
+
+    request.setRawHeader("User-Agent", "curl/7.58.0");
+    //request.setRawHeader("User-Agent", userName.toLocal8Bit());
+
+    if (tokenOrPassword.size() == 0)        // not authorized request
+    {
+

```

Comment:
?

Reply:
it is not for to do
it is for clarification code